src/Database/Relations/HasOneOrMany.php

```diff
@@ -57,7 +57,10 @@ public function add(Model $model, $sessionKey = null)
     {
         if ($sessionKey === null) {
             $model->setAttribute($this->getForeignKeyName(), $this->getParentKey());
-            $model->save();
+
+            if (!$model->exists || $model->isDirty()) {
```

Comment:
@bennothommo @daftspunk wouldn't this prevent existing models from being added to the relationship?

```php
$comment = new Comment();
$comment->save();

$post->add($comment);
```

Reply:
@LukeTowers I haven't tested that particular case, but I think the line above it, setting the related attribute to the new parent model, would make the child model dirty, so it would still save.

Reply:
Ah, right. I got confused by thinking that `||` was being processed like `&&`, where it would short circuit if the first value was false, but it's actually the opposite so this is fine. Thanks for the response!